src/Base/Exception.h

```diff
@@ -33,6 +33,13 @@
 #include "FileInfo.h"
 #include "BaseClass.h"
 
+/* MACROS FOR THROWING EXCEPTIONS */
+
+#define THROW(exception) throw Base::exception(__FILE__,__LINE__,__PRETTY_FUNCTION__);
+#define THROWM(exception, message) throw Base::exception(message,__FILE__,__LINE__,__PRETTY_FUNCTION__);

```

Comment:
> Are you interested in using these macros for your exceptions at this time? If yes, should I move them out of Exception.h? If yes, where to?

Yes I am interested in using these macros. in src/App/Containers/Exceptions.h, I defined quite a few specific exceptions, with python twins, and they derive from Base::Exception. But they are in App namespace.

Your macros defined in Base/Exception.h are no problem to use. I see no reason to move them elsewhere. Just remove the "Base::" hard-coded into the macro, so that we can supply our own namespace (or lack of it, if `using namespace xxx`)